### DISCUSSION TITLE
List in RpcUrls type is redundant

### DISCUSSION THREAD

**Discussion by jmrossy** [wevm/wagmi#1505] [General]
The @wagmi/chains 0.9.0 update introduced a new shape for the Rpc urls in the Chain objects:
```
    /** Collection of RPC endpoints */
    rpcUrls: {
        [key: string]: RpcUrls;
        default: RpcUrls;
    };

[...]

declare type RpcUrls = {
    http: string[];
    webSocket?: string[];
};
```

Each protocol type is a **list** of urls, but the the rpcUrls prop in `Chain` is already a object, meaning there will be multiple entries. So why force a list? Indeed, none of the Chain configs in Wagmi's own consts use more than one value in these lists: https://github.com/wagmi-dev/references/tree/e987fce4441bfa5906a6b9b0b68e580665418651/packages/chains/src

Not a huge deal, just feels like an unnecessary obstruction.

**Comment by jxom** [comment]
It is defined as a list as a particular group (key of `RpcUrls`) could have multiple URLs, and it generally designed for flexibility/future proofing – this will be useful when wagmi starts to have enhanced compatibilty with [Fallback Providers](https://docs.ethers.org/v5/api/providers/other/#FallbackProvider).

For example, a chain could have multiple **public** RPC URLs, and wagmi (in the future), may want to apply those URLs to the Fallback Provider.

**Reply by jmrossy** [reply]
Thanks for the reply!
When would it be preferable for the extra RPC URLs to be appended to the list under an existing key (e.g. `default`), instead of under their own key with a more descriptive name?

**Reply by jxom** [reply]
It would be preferable for the extra URLs to be appended to the list when it belongs to that group.

Examples could be:
- A **public** (`rpcUrls.default`) group of RPC URLs – that a consuming library (e.g. wagmi) may want to roundrobin/create fallbacks,
- An RPC Provider like bloXroute (e.g. `rpcUrls.bloxroute`) could have multiple RPC URLs depending on location. [See here](https://docs.bloxroute.com/introduction/cloud-api-ips). A consuming library may want to have control over these URLs, and pick the fastest one for the user.

**Reply by jmrossy** [reply]
Okay, thanks for clarifying :)
